tests/tasks/teardown/awscli-eks.yaml

```diff
@@ -26,4 +26,10 @@ spec:
       if [ -n "$(params.endpoint)" ]; then
         ENDPOINT_FLAG="--endpoint $(params.endpoint)"
       fi
+
+      for i in `aws eks list-nodegroups --cluster-name $(params.cluster-name) $ENDPOINT_FLAG | jq -r '.nodegroups[]'`;
+      do
+          aws eks delete-nodegroup --nodegroup-name $i --cluster-name $(params.cluster-name) $ENDPOINT_FLAG;
+          aws eks wait nodegroup-deleted --nodegroup-name $i --cluster-name $(params.cluster-name) $ENDPOINT_FLAG;
```

Comment:
This may be good enough for now. But sometimes these calls may fail (for e.g timeout while waiting for nodegroup to go down). What's the behavior in that case? Will we drop the `finally` block after single attempt and go ahead with the next run anyway?

If there's an "initialization" or "pre-flight check" option, might be worth adding that too as a blocker before actually starting the test logic.

Reply:
Yes, there is a possibility where Finally step can fail during server side (eks api) failures. Probability of such failures are less, and I have thought about that corner case, our defense to that failures is we are always creating clusters with same `name`, so we can only have one cluster per kit environment at any given point in time unlike before because, EKS doesn't allow user to create cluster with same `name` twice in same region. However we may want to be notified when we see server side failures during  such runs, which is why I have added this backlog issue for us  - https://github.com/awslabs/kubernetes-iteration-toolkit/issues/230

Reply:
Also, if we fail in the `finally` call and leak resources, in the next iteration we will reconcile on an existing cluster with the cluster creation task and `finally` will try to clean again. As Harish mentioned we will need to be notified when `finally` fails, so we need an alarm.